client/client-multi/client-java-core/src/test/java/org/kaaproject/kaa/client/context/FlexibleExecutorContextTest.java

```diff
@@ -0,0 +1,201 @@
+/*
+ * Copyright 2014-2016 CyberVision, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.kaaproject.kaa.client.context;
+
+import org.junit.Assert;
+import org.junit.Test;
+
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.ScheduledThreadPoolExecutor;
+import java.util.concurrent.ThreadPoolExecutor;
+import java.util.concurrent.TimeUnit;
+
+public class FlexibleExecutorContextTest {
+
+  private FlexibleExecutorContext executorContext;
+
+  private static final int DEFAULT_CORE_POOL_SIZE = 1;
+  private static final int DEFAULT_MAX_POOL_SIZE = FlexibleExecutorContext.getDefaultMaxThreads();
+  private static final int DEFAULT_MAX_THREAD_IDLE_MILLISECONDS = FlexibleExecutorContext.getDefaultMaxThreadIdleMilliseconds();
+
+  private static final int CUSTOM_MAX_LIFE_CYCLE_THREADS_IDLE_MILLISECONDS = 40;
+  private static final int CUSTOM_MAX_API_THREADS_IDLE_MILLISECONDS = 50;
+  private static final int CUSTOM_MAX_CALLBACK_THREADS_IDLE_MILLISECONDS = 60;
+
+  private static final int CUSTOM_LIFE_CYCLE_THREAD_COUNT = 4;
+  private static final int CUSTOM_API_THREAD_COUNT = 5;
+  private static final int CUSTOM_CALLBACK_THREAD_COUNT = 6;
+  private static final int CUSTOM_SCHEDULED_THREAD_COUNT = 7;
+
+  @Test
+  public void testConstructorWithoutParameters() throws Exception {
+    executorContext = new FlexibleExecutorContext();
+    executorContext.init();
+
+    ExecutorService executorService;
+    ThreadPoolExecutor threadPoolExecutor;
+    long corePoolSize, maxPoolSize, maxThreadIdleMilliseconds;
```

Comment:
Use **int** type as in constants.
Declare every variable in separate line. No grouping reccomended.